### PR TITLE
remove unused temporary variables in code_warntype

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: base/interactiveutil.jl [hunk 1]
```diff
@@ -374,15 +374,15 @@ function code_warntype(io::IO, f, @nospecialize(t))
...
-            print(emph_io, "  ", slotnames[i])
             if used_slotids[i]
+                print(emph_io, "  ", slotnames[i])
                 if isa(src.slottypes, Array)
                     show_expr_type(emph_io, src.slottypes[i], true)
                 end
-            else
-                print(emph_io, " <optimized out>")
+                print(emph_io, '\n')
+            elseif !('#' in slotnames[i] || '@' in slotnames[i])
```
Comment (yuyichao) [inline base/interactiveutil.jl]: Note that slot names won't have automatically generated `@` in them.

**Review by yuyichao — COMMENTED** [review]
I didn't suggest this originally since the full list provides information about the slot number (mainly useful for connecting `code_warntype` output to the `jl_` output of the expressions in the debugger). There are less straight forward ways to obtain this information so while I prefer keeping it or at least making slot numbers inferable from `code_warntype` output I'm kind of ok with removing these.....

**Comment by KristofferC** [comment]
If this is needed for some low-level debugging there could be e.g. a flag that enables printing these again. I just don't think that printing them by default is necessary since they are likely of little value to most people who just want to check the type stability of their code.

**Comment by KristofferC** [comment]
I think this one would be nice to have on 0.6. @ararslan

**Comment by ararslan** [comment]
I'm not sure that this is backport eligible since it's technically a feature rather than a bugfix, right? Not sure though. If it's legit to backport then I'd be happy to do so.

**Comment by KristofferC** [comment]
Alright, no big deal.

**Pull request merged**